js/models/messages.js

```diff
@@ -561,14 +561,17 @@
           }
         }
         message.set({
-          schemaVersion: dataMessage.schemaVersion,
+          attachments: dataMessage.attachments,
           body: dataMessage.body,
           conversationId: conversation.id,
-          attachments: dataMessage.attachments,
-          quote: dataMessage.quote,
           decrypted_at: now,
-          flags: dataMessage.flags,
           errors: [],
+          flags: dataMessage.flags,
+          hasAttachments: dataMessage.hasAttachments,
+          hasFileAttachments: dataMessage.hasFileAttachments,
+          hasVisualMediaAttachments: dataMessage.hasVisualMediaAttachments,
+          quote: dataMessage.quote,
+          schemaVersion: dataMessage.schemaVersion,
```

Comment:
Hope we can at some point implement a `Message.parse` function that spits out only valid `Message` objects with whitelisted properties, correct types, and reasonable defaults, and simply pass it all in here…